test/unit/compute_resource_test.rb

```diff
@@ -271,4 +271,20 @@ def setup
       assert(cr.valid?, 'ComputeResource can have spaces in name')
     end
   end
+
+  describe "host_interfaces_attrs" do
+    before do
+      @cr = compute_resources(:mycompute)
+    end
+
+    test "only physical interfaces are added to the compute attributes" do
+      physical_nic = FactoryGirl.build(:nic_base, :virtual => false,
+                                       :compute_attributes => { :virtual => false })
+      virtual_nic = FactoryGirl.build(:nic_base, :virtual => true,
+                                       :compute_attributes => { :virtual => true })
+      host = FactoryGirl.build(:host, :interfaces => [physical_nic, virtual_nic])
+      nic_attributes = @cr.host_interfaces_attrs(host)
+      assert_equal 1, nic_attributes.count { |k, v| v.present? }
```

Comment:
could you please add an assertion that the found interface is physicial? (e.g. add it some identifier few lines above and check it here)